src/server/middleware/auth.ts

```diff
@@ -449,3 +482,28 @@ const hashRequestBody = (req: FastifyRequest): string => {
     .update(JSON.stringify(req.body), "utf8")
     .digest("hex");
 };
+
+/**
+ * Check if the request IP is in the allowlist.
+ * Fetches cached config if available.
+ * env.TRUST_PROXY is used to determine if the X-Forwarded-For header should be trusted.
```

Comment:
The comment on TRUST_PROXY should be near where the var is actually used, in `server/index.ts`